tests/test_taar_lite_guidguid.py

```diff
@@ -154,6 +155,64 @@ def test_addon_keying():
     assert taar_lite_guidguid.key_all(MOCK_KEYED_ADDONS) == EXPECTED_ADDON_INSTALLATIONS
 
 
+def test_get_addons_per_client(spark, df_equals):
+    def make_addon_meta(addon_id):
+        return Row(
+            **dict(
+                addon_id=addon_id,
+                is_system=False,
+                type="extension",
+                user_disabled=False,
+                app_disabled=False,
+                foreign_install=False,
+            )
+        )
+
+    MOCK_USERS_DF = [
+        Row(
+            client_id="client-1",
+            active_addons=[
+                make_addon_meta("test-guid-1"),
+                make_addon_meta("test-guid-2"),
+                make_addon_meta("test-guid-3"),
+            ],
+        ),
+        Row(
+            client_id="client_2",
+            active_addons=[
+                make_addon_meta("test-guid-1"),
+                make_addon_meta("test-guid-3"),
+            ],
+        ),
+        Row(
+            client_id="client_3",
+            active_addons=[
+                make_addon_meta("test-guid-1"),
+                make_addon_meta("test-guid-4"),
+            ],
+        ),
+        Row(
+            client_id="client_4",
+            active_addons=[
+                make_addon_meta("test-guid-2"),
+                make_addon_meta("test-guid-5"),
+                make_addon_meta("test-guid-6"),
+            ],
+        ),
+        Row(client_id="client_5", active_addons=[make_addon_meta("test-guid-1")]),
+    ]
+
+    test_users_df = spark.createDataFrame(MOCK_USERS_DF)
+
+    mock_broadcast_amo_whitelist = Mock()
+    mock_broadcast_amo_whitelist.value = Mock()
+    mock_broadcast_amo_whitelist.value.__contains__ = lambda self, x: True
+
+    taar_lite_guidguid.get_addons_per_client(
+        mock_broadcast_amo_whitelist, test_users_df
+    )
```

Comment:
```suggestion
    ).count()
```

You should materialize this dataframe to make sure it's working properly.